pythonforandroid/bootstraps/sdl2/build/jni/Application.mk

```diff
@@ -5,3 +5,4 @@
 
 # APP_ABI := armeabi armeabi-v7a x86
 APP_ABI := $(ARCH)
+APP_PLATFORM := android-21
```

Comment:
This causes sdl2 recipe to fail. We can comment this line to bypass the error...as a temporary hack in order to make further tests, **NOT AS A FINAL SOLUTION**.

`# APP_PLATFORM := android-21`


Note: This already has been informed by @inclement into the "to do" section : "add proper APP_PLATFORM setting functionality"


Here are the log of the failure:

```[INFO]:    -> directory context /home/opacam/.buildozer-builds/android/platform/build/build/bootstrap_builds/sdl2-python3/jni
[DEBUG]:   -> running ndk-build V=1
[DEBUG]:        Android NDK: APP_PLATFORM set to unknown platform: 21.
[DEBUG]:        /media/DEVEL/Android/android-ndk-r17b/build/core/setup-app-platform.mk:135: *** Android NDK: Aborting    .  Stop.
Exception in thread background thread for pid 31274:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 1540, in wrap
    fn(*args, **kwargs)
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 2459, in background_thread
    handle_exit_code(exit_code)
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 2157, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 815, in handle_command_exit_code
    raise exc
ErrorReturnCode_2: 

  RAN: /media/DEVEL/Android/android-ndk-r17b/ndk-build V=1

  STDOUT:
Android NDK: APP_PLATFORM set to unknown platform: 21.    
/media/DEVEL/Android/android-ndk-r17b/build/core/setup-app-platform.mk:135: *** Android NDK: Aborting    .  Stop.


  STDERR:


Traceback (most recent call last):
  File "/home/opacam/.local/bin/p4a", line 11, in <module>
    load_entry_point('python-for-android', 'console_scripts', 'p4a')()
  File "/home/opacam/Devel/python-for-android/pythonforandroid/toolchain.py", line 981, in main
    ToolchainCL()
  File "/home/opacam/Devel/python-for-android/pythonforandroid/toolchain.py", line 518, in __init__
    getattr(self, args.subparser_name.replace('-', '_'))(args)
  File "/home/opacam/Devel/python-for-android/pythonforandroid/toolchain.py", line 148, in wrapper_func
    build_dist_from_args(ctx, dist, args)
  File "/home/opacam/Devel/python-for-android/pythonforandroid/toolchain.py", line 192, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx)
  File "/home/opacam/Devel/python-for-android/pythonforandroid/build.py", line 579, in build_recipes
    recipe.build_arch(arch)
  File "/home/opacam/Devel/python-for-android/pythonforandroid/recipes/sdl2/__init__.py", line 41, in build_arch
    shprint(sh.ndk_build, "V=1", _env=env)
  File "/home/opacam/Devel/python-for-android/pythonforandroid/logger.py", line 176, in shprint
    for line in output:
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 863, in next
    self.wait()
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 792, in wait
    self.handle_command_exit_code(exit_code)
  File "/home/opacam/.local/lib/python2.7/site-packages/sh.py", line 815, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_2: 

  RAN: /media/DEVEL/Android/android-ndk-r17b/ndk-build V=1

  STDOUT:
Android NDK: APP_PLATFORM set to unknown platform: 21.    
/media/DEVEL/Android/android-ndk-r17b/build/core/setup-app-platform.mk:135: *** Android NDK: Aborting    .  Stop.```

Reply:
I don't know what caused this for you, does your NDK not have platform-21?